plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/timeline/HudiActiveTimeline.java

```diff
@@ -85,7 +85,11 @@ public Optional<byte[]> readCompactionPlanAsBytes(HudiInstant instant)
 
     private Location getInstantFileNamePath(String fileName)
     {
-        return Location.of(fileName.contains(SCHEMA_COMMIT_ACTION) ? metaClient.getSchemaFolderName() : metaClient.getMetaPath().path()).appendPath(fileName);
+        Location metaPath = metaClient.getMetaPath();
+        if (fileName.contains(SCHEMA_COMMIT_ACTION)) {
+            return metaPath.appendPath(HudiTableMetaClient.SCHEMA_FOLDER_NAME);
```

Comment:
Is `.appendPath(fileName)` missing? Do existing tests cover this condition?

Reply:
I have added the `appendPath(fileName)` to the schema commit path of the condition, it was clearly missing from there. 

In order to test this, we need a second table with at least one `schemacommit` action in the Hudi timeline. Will work on preparing that as well in this patchset.

Reply:
Please note my last comment below regarding this.